Client/Assets/SessionRestore.html

```diff
@@ -0,0 +1,22 @@
+<!DOCTYPE html>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+ - License, v. 2.0. If a copy of the MPL was not distributed with this
+ - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+ <html>
+ <script>
+    (function () { 
+        var index = document.location.href.search("history");
+        var sessionRestoreComponents = JSON.parse(unescape(document.location.href.substring(index + "history=".length)))
+        var urlList = sessionRestoreComponents['history']
+        var currentPage = sessionRestoreComponents['currentPage']
+        history.replaceState({}, '', '/errors/error.html?url=' + escape(urlList[0]))
+        for (var i = 1; i < urlList.length; i++) {
+            history.pushState({}, '', '/errors/error.html?url=' + escape(urlList[i]))
+        }
+        history.go(currentPage)
+        setTimeout(function () {
```

Comment:
`// Finally, reload the page to trigger the error redirection, which will load the actual URL. For some reason (maybe a WebKit bug?), document.location still points to SessionRestore.html at this point, so wait until the next tick when the location points to the correct index and URL.`